can/util.py

```diff
@@ -233,25 +233,6 @@ def _create_bus_config(config: Dict[str, Any]) -> typechecking.BusConfig:
     if "data_bitrate" in config:
         config["data_bitrate"] = int(config["data_bitrate"])
 
-    # Create bit timing configuration if given
-    timing_conf = {}
-    for key in (
-        "f_clock",
-        "brp",
-        "tseg1",
-        "tseg2",
-        "sjw",
-        "nof_samples",
-        "btr0",
-        "btr1",
-    ):
-        if key in config:
-            timing_conf[key] = int(str(config[key]), base=0)
-            del config[key]
```

Comment:
Have you tried just removing this line with `del` instead of everything?

Reply:
For my specific issue with the PCAN interface, it's actually line 252 that's causing the immediate problem (although the `del` would also be an issue further down the road). Since it detects that `f_clock` is present in `kwargs`, it automatically tries to create the `timing_conf` and simply assumes that the config must also contain a `bitrate` key in line 252. This is not always the case for PCAN.

There are multiple reasons for me suggesting the removal of this piece of code:
* It causes issues with at least two interfaces
* It doesn't look like the code is actively used by any interface or anyone has picked up on it since it was merged
* The contract between `Bus.__new__` and the interface implementations is very weak. It basically says: If you name your parameters in the correct way, we'll automagically create the `BitTiming` instance for you. But only for Classic CAN, not CAN-FD.
* The author itself stated in his commit that it (obviously) doesn't support CAN-FD.